sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -133,129 +192,280 @@ public void E2E_ThreadCreateUpdateGetDelete_MemberAddUpdateRemove_MessageGetSend
-
-            #region Snippet:Azure_Communication_Chat_Tests_E2E_GetReadReceipts
-            Pageable<ReadReceipt> readReceipts = chatThreadClient.GetReadReceipts();
-            #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetReadReceipts
-            var readReceiptsCount = readReceipts.Count();
-
             #region Snippet:Azure_Communication_Chat_Tests_E2E_DeleteChatThread
             chatClient.DeleteChatThread(threadId);
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_DeleteChatThread
 
             //assert
             Assert.AreEqual(updatedTopic, chatThread.Topic);
-            Assert.AreEqual(2, chatThread.Members.Count);
+            Assert.AreEqual(3, chatThread.Members.Count);
             Assert.AreEqual(messageContent, message.Content);
             Assert.AreEqual(displayNameMessage, message.SenderDisplayName);
+            Assert.AreEqual(updatedMessageContent, actualUpdateMessage.Value.Content);
             Assert.AreEqual(ChatMessagePriority.High, message.Priority);
+
+            Assert.AreEqual(messageContent2, message2.Content);
+            Assert.AreEqual(messageContent3, message3.Content);
+            Assert.AreEqual(messageContent4, message4.Content);
+            Assert.AreEqual(messageContent5, message5.Content);
+            Assert.AreEqual(messageContent6, message6.Content);
+
             Assert.AreEqual(2, threadsCount);
-            Assert.AreEqual(5, getMessagesCount); //Including all types
-            Assert.AreEqual(updatedMessageContent, actualUpdateMessage.Value.Content);
+            Assert.AreEqual(8, getMessagesCount); //Including all types : 5 text message, 3 control messages
+            Assert.AreEqual(3, getMessagesCount2); //Including all types : 1 text message, 2 control messages
+
             Assert.IsTrue(deletedChatMessage.DeletedOn.HasValue);
-            Assert.AreEqual(2, chatThreadMembersCount);
-            Assert.AreEqual(3, chatThreadMembersAfterOneAddedCount);
-            Assert.AreEqual(2, chatThreadMembersAfterOneDeletedCount);
+            Assert.AreEqual(3, chatThreadMembersCount);
+            Assert.AreEqual(4, chatThreadMembersAfterOneAddedCount);
+            Assert.AreEqual(3, chatThreadMembersAfterOneDeletedCount);
             Assert.AreEqual((int)HttpStatusCode.OK, typingNotificationResponse.Status);
-            // TODO: Commenting out the assert below for now as it is flakey due to server-side delay; currently in investigation
-            // Assert.AreEqual(1, readReceiptsCount);
         }
 
+        [SyncOnly]
+        [Test]
+        [PlaybackOnly("Message and ReadReceipt storage uses eventual consistency. Tests to get readreceipts requires delays")]
+        public void ReadReceiptGS()
+        {
+            //arr
+            Console.WriteLine($"ReadReceiptGS Running on RecordedTestMode : {Mode}");
+            CommunicationIdentityClient communicationIdentityClient = CreateInstrumentedCommunicationIdentityClient();
```

Comment:
Imo using `var` throughout the code would make it more readable, not sure what Azure conventions say.